src/uct/ib/ud/base/ud_iface.c

```diff
@@ -199,6 +200,20 @@ uct_ud_iface_create_qp(uct_ud_iface_t *self, const uct_ud_iface_config_t *config
     qp_init_attr.cap.max_recv_sge    = 1;
     qp_init_attr.cap.max_inline_data = config->super.tx.min_inline;
 
+    if (qp_init_attr.cap.max_inline_data > dev->max_inline_data) {
+        ucs_diag("create QP: TX inline: using device cap %uB (configured %uB)",
+                 dev->max_inline_data, qp_init_attr.cap.max_inline_data);
+
+        qp_init_attr.cap.max_inline_data = dev->max_inline_data;
+    }
+
+    if (qp_init_attr.cap.max_send_sge > dev->max_sq_sge) {
+        ucs_diag("create QP: max_sq_sge: using device cap %d (configured %u+1)",
+                 dev->max_sq_sge, qp_init_attr.cap.max_send_sge - 1);
```

Comment:
why `+1` and `-1`?

Reply:
configuration assumes max_sge without net header, this is to try to make it explicit, i guess dumping actual values is simpler eventually

Reply:
imo it is less clear than just printing actual values, but it is up to you

Reply:
fixed it